yarn-project/circuits.js/src/structs/avm/avm.ts

```diff
@@ -2,9 +2,82 @@ import { type Fr } from '@aztec/foundation/fields';
 import { serializeToBuffer } from '@aztec/foundation/serialize';
 import { type FieldsOf } from '@aztec/foundation/types';
 
-// TODO(https://github.com/AztecProtocol/aztec-packages/issues/6774): add public inputs.
+import { type PublicCircuitPublicInputs } from '../public_circuit_public_inputs.js';
+
+export class AvmHint {
+  constructor(public readonly key: Fr, public readonly value: Fr) {}
+
+  /**
+   * Serializes the inputs to a buffer.
+   * @returns - The inputs serialized to a buffer.
+   */
+  toBuffer() {
+    return serializeToBuffer(...AvmHint.getFields(this));
+  }
+
+  /**
+   * Serializes the inputs to a hex string.
+   * @returns The instance serialized to a hex string.
+   */
+  toString() {
+    return this.toBuffer().toString('hex');
+  }
+
+  /**
+   * Extracts fields from an instance.
+   * @param fields - Fields to create the instance from.
+   * @returns An array of fields.
+   */
+  static getFields(fields: FieldsOf<AvmHint>) {
+    return [fields.key, fields.value] as const;
+  }
+}
+
+export class AvmExecutionHints {
+  constructor(
+    public readonly storageValues: AvmHint[],
+    public readonly noteHashExists: AvmHint[],
+    public readonly nullifierExists: AvmHint[],
+    public readonly l1ToL2MessageExists: AvmHint[],
+  ) {}
+
+  /**
+   * Serializes the inputs to a buffer.
+   * @returns - The inputs serialized to a buffer.
+   */
+  toBuffer() {
+    return serializeToBuffer(...AvmExecutionHints.getFields(this));
```

Comment:
The arrays are variable-size. Could you confirm that the serialization encodes the size? Otherwise the CPP side will not be able to deserialize.

Reply:
Used `serializeArrayOfBufferable` for each hint array instead